src/Tickets/Commerce/Promoter_Observer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace TEC\Tickets\Commerce;
+
+use Tribe\Tickets\Promoter\Triggers\Contracts\Attendee_Model;
+use Tribe\Tickets\Promoter\Triggers\Models\Attendee;
+
+class Promoter_Observer {
+
+	/**
+	 * @since 4.12.0
+	 *
+	 * @var Tribe__Tickets__Promoter__Observer $observer ET Observer reference.
+	 */
+	private $observer;
+
+	/**
+	 * Constructor.
+	 *
+	 */
+	public function __construct() {
+		$this->observer = tribe( \Tribe__Tickets__Promoter__Observer::class );
+		$this->hook();
+	}
+
+	/**
+	 * Attach hooks for trigger messages.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		add_action( 'tec_tickets_commerce_flag_action_generated_attendee', [ $this, 'attendee_created' ], 10, 5 );
+		add_action( 'event_tickets_checkin', [ $this, 'checkin' ], 10, 2 );
+	}
```

Comment:
Let's add a hook to `tec_tickets_commerce_ticket_deleted` to notify Promoter when a ticket is deleted. We don't need to do one for a ticket save, because that should be handled by [`Tribe__Tickets__Promoter__Observer`](https://github.com/the-events-calendar/event-tickets/blob/master/src/Tribe/Promoter/Observer.php#L62).

```suggestion

		add_action( 'tec_tickets_commerce_flag_action_generated_attendee', [ $this, 'attendee_created' ], 10, 5 );
		add_action( 'tec_tickets_commerce_ticket_deleted', tribe_callback( 'tickets.promoter.observer', 'notify_event_id' ) );
		add_action( 'event_tickets_checkin', [ $this, 'checkin' ], 10, 2 );
	}
```